src/osgDB/Registry.cpp

```diff
@@ -1171,7 +1168,7 @@ ReaderWriter::ReadResult Registry::read(const ReadFunctor& readFunctor)
 
             options->setDatabasePath(archiveName);
 
-            osg::ref_ptr<ReadFunctor> rf(readFunctor.cloneType(fileName, options.get()));
+            std::auto_ptr<ReadFunctor> rf(readFunctor.cloneType(fileName, options.get()));
```

Comment:
This is a strange change. Why was it done? I am not seeing the same change in makOsg 3.4

Reply:
Or at least it seems like we should take the 3.6 change in this case instead of using the old 3.4 way.